crypto/asn1/bio_ndef.c

```diff
@@ -121,9 +121,10 @@ static int ndef_prefix(BIO *b, unsigned char **pbuf, int *plen, void *parg)
     *pbuf = p;
     derlen = ASN1_item_ndef_i2d(ndef_aux->val, &p, ndef_aux->it);
 
-    if (!*ndef_aux->boundary)
+    if (!*ndef_aux->boundary){
```

Comment:
This is still not correct yet. Move this check near the top of the function - before p is allocated.